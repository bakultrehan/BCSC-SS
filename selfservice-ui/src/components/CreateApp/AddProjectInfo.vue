/** * Project information */
<template>
  <v-card class="mx-auto" style="max-width: 80%;">
    <v-card class="mx-auto">
      <v-app-bar dark color="#003366">
        <v-btn
          icon
          @click="$router.push('/projectinfo/')"
          aria-label="Back Button"
        >
          <v-icon>mdi-arrow-left</v-icon>
        </v-btn>
        <v-toolbar-title>Project information</v-toolbar-title>
        <v-spacer></v-spacer>
      </v-app-bar>
      <v-form ref="form" v-model="form">
        <v-container>
          <v-row dense>
            <v-col cols="12" md="12">
              <v-card class="pa-4 pt-6">
                <v-card-title class="headline"
                  >Project information</v-card-title
                >
                <v-card-subtitle class="text-left"
                  >Tell us about your Project</v-card-subtitle
                >
                <Input
                  v-model="organizationName"
                  counter="200"
                  label="Organization name"
                  type="text"
                  :rules="[rules.required, rules.length(2)]"
                />
                <Input
                  v-model="projectName"
                  counter="100"
                  label="Project name"
                  type="text"
                  :rules="[rules.required, rules.length(2)]"
                />
                <TextArea
                  v-model="description"
                  label="Description"
                  type="text"
                  :rules="[rules.required]"
                />
              </v-card>
            </v-col>

            <v-col cols="12">
              <v-card class="pa-4 pt-6">
                <v-card-title class="headline">Project Roles</v-card-title>
                <v-card-subtitle class="text-left"
                  >Tell us about your role in this project</v-card-subtitle
                >
                <v-radio-group v-model="myRole" row>
                  I am
                  <v-radio label="Developer" value="1"></v-radio>
                  <v-radio label="Manager" value="2"></v-radio>
                  <v-radio label="CTO" value="3"></v-radio>For this project
                </v-radio-group>
              </v-card>
            </v-col>
            <!-- <v-form ref="form" v-model="form" class="pa-4 pt-6"> -->
            <v-col cols="12" sm="6" v-if="myRole !== '2'">
              <ProjectUsers
                :userDetails="managerDetails"
                :rules="rules"
                title="Manager"
              />
            </v-col>
            <v-col cols="12" sm="6" v-if="myRole !== '3'">
              <ProjectUsers
                :userDetails="ctoDetails"
                :rules="rules"
                title="CTO"
              />
            </v-col>
            <v-col cols="12" sm="6" v-if="myRole !== '1'">
              <ProjectUsers
                :userDetails="developerDetails"
                :rules="rules"
                title="Developer"
              />
            </v-col>

            <v-col cols="12">
              <v-card flat>
                <v-divider></v-divider>
                <v-card-actions>
                  <!-- <v-btn text @click="$refs.form.reset()">Clear</v-btn> -->
                  <v-spacer></v-spacer>
                  <Button
                    :disabled="!form"
                    :loading="isLoading"
                    class="white--text"
                    color="indigo accent-4"
                    depressed
                    @click="addProjectInfo"
                    >Next</Button
                  >
                </v-card-actions>
              </v-card>
            </v-col>
          </v-row>
        </v-container>
      </v-form>
    </v-card>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import { ProjectUserModel, ProjectInfoModel } from '@/models/ProjectInfoModel';
import Input from '@/Atomic/Input/Input.vue';
import TextArea from '@/Atomic/TextArea/TextArea.vue';
import Button from '@/Atomic/Button/Button.vue';
import ProjectUsers from './ProjectUsers.vue';
import validationRules from '@/config/validationRules';

const ProjectInfoModule = namespace('ProjectInfoModule');

@Component({ components: { Input, TextArea, Button, ProjectUsers } })
export default class AddProjectInfo extends Vue {
  @Prop({ default: '' })
  public id!: string;
  @Prop({ default: '' })
  public action!: string;

  @ProjectInfoModule.Getter('successStatus') public successStatus!: boolean;
  @ProjectInfoModule.Getter('errorStatus') public errorStatus!: boolean;
  @ProjectInfoModule.Action('addProjectInfo') public addProjectInfoStore!: any;
  @ProjectInfoModule.Action('loadProjectInfo')
  public loadProjectInfo!: any;
  @ProjectInfoModule.Getter('getSingleProjectInfo')
  public getSingleProjectInfo!: any;
  @ProjectInfoModule.Action('updateProjectInfo')
  public updateProjectInfoStore!: any;
  @ProjectInfoModule.Action('loadSingleProjectInfo')
  public loadSingleProjectInfo!: any;

  public form: boolean = false;
  private isLoading: boolean = false;
  private organizationName: string = '';
  private projectName: string = '';
  private description: string = '';
  private myRole: string = '1';
  private developerDetails?: undefined;
  private managerDetails?: ProjectUserModel = {
    email: '',
    phone: '',
    firstName: '',
    lastName: ''
  };
  private ctoDetails?: ProjectUserModel = {
    email: '',
    phone: '',
    firstName: '',
    lastName: ''
  };

  private isEditmode: boolean = false;
  /* istanbul ignore next */
  private rules = validationRules;

  @Watch('getSingleProjectInfo')
  private ongetSingleProjectInfoChanged(val: any) {
    this.updteEdit(val);
  }

  private addProjectInfo() {
    const data: ProjectInfoModel = {
      organizationName: this.organizationName,
      projectName: this.projectName,
      description: this.description,
      myRole: this.myRole,
      developerDetails: this.developerDetails,
      managerDetails: this.managerDetails,
      ctoDetails: this.ctoDetails
    };

    if (this.isEditmode) {
      data.id = this.id;
      this.updateProjectInfoStore(data);
    } else {
      this.addProjectInfoStore(data);
    }
    // (this.$refs.form as HTMLFormElement).reset();
    this.$router.push('/Project/technical/');
  }

  private updteEdit(val: any) {
    this.organizationName = val.organizationName;
    this.projectName = val.projectName;
    this.description = val.description;
    this.myRole = val.myRole;
    this.developerDetails = val.developerDetails;
    this.managerDetails = val.managerDetails;
    this.ctoDetails = val.ctoDetails;
    this.id = val.id;
    this.isEditmode = true;
  }

  private mounted() {
    this.isEditmode = false;
    if (this.action && this.action === 'edit' && this.id !== '') {
      this.isEditmode = true;
      this.loadSingleProjectInfo(this.id);
    }
  }
  private input(value: string) {
    // console.log('value', value);
  }
}
</script>
