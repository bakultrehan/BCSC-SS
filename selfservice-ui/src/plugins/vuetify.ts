import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
