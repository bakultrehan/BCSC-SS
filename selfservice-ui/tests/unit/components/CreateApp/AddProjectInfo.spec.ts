import { shallowMount, createLocalVue } from '@vue/test-utils';
import AddProjectInfo from '@/components/CreateApp/AddProjectInfo.vue';
import vuetify from 'vuetify';
import Vuex from 'vuex';

describe('AddProjectInfo.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);
    localVue.use(Vuex);
    const store = new Vuex.Store({
      modules: {
        ProjectInfoModule: {
          namespaced: true,
          state: {},
          getters: {
            getSingleProjectInfo: jest.fn(),
            isLoggedin: jest.fn()
          }
        }
      }
    });
    wrapper = shallowMount(AddProjectInfo, {
      localVue,
      store,
      mocks: { $t: jest.fn(() => {}) } // tslint:disable-line
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
