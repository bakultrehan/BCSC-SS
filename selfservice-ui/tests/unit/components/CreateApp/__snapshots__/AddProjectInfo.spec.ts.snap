// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProjectInfo.vue renders edit props when passed 1`] = `
<v-card-stub
  class="mx-auto outer-card"
  loaderheight="4"
  tag="div"
>
  <v-card-stub
    class="mx-auto"
    loaderheight="4"
    tag="div"
  >
    <v-app-bar-stub
      class="bc-subtitle"
      dark="true"
      extensionheight="48"
      src=""
      tag="header"
      tile="true"
      value="true"
    >
      <v-btn-stub
        activeclass=""
        aria-label="Back Button"
        icon="true"
        tag="button"
        type="button"
      >
        <v-icon-stub>
          mdi-arrow-left
        </v-icon-stub>
      </v-btn-stub>
       
      <v-toolbar-title-stub>
        
        
      
      </v-toolbar-title-stub>
       
      <v-spacer-stub />
    </v-app-bar-stub>
     
    <v-form-stub>
      <v-container-stub
        tag="div"
      >
        <v-row-stub
          dense="true"
          tag="div"
        >
          <v-col-stub
            cols="12"
            md="12"
            tag="div"
          >
            <v-card-stub
              class="pa-8 pt-6 ma-3"
              loaderheight="4"
              tag="div"
            >
              <v-card-subtitle-stub
                class="text-left bc-padding-left-0 page-info"
              />
               
              <input-stub
                data-test-id="input-org-name"
                helptext=""
                label=""
                outlined="true"
                rules="v => !!v || 'This field is required',v => (v || '').length >= len || 'Invalid character length, required ' + len,v => (v || '').length <= len || 'Invalid character length, max length ' + len"
                type="text"
                value=""
              />
               
              <input-stub
                data-test-id="input-project-name"
                helptext=""
                label=""
                outlined="true"
                rules="v => !!v || 'This field is required',v => (v || '').length >= len || 'Invalid character length, required ' + len,v => (v || '').length <= len || 'Invalid character length, max length ' + len"
                type="text"
                value=""
              />
               
              <textarea-stub
                counter=""
                data-test-id="text-project-description"
                helptext=""
                label=""
                outlined=""
                rules="v => !!v || 'This field is required'"
                type="text"
                value=""
              />
            </v-card-stub>
          </v-col-stub>
           
          <v-col-stub
            cols="12"
            tag="div"
          >
            <v-card-stub
              flat="true"
              loaderheight="4"
              tag="div"
            >
              <v-divider-stub />
               
              <v-card-actions-stub>
                <v-spacer-stub />
                 
                <!---->
                 
                <button-stub
                  class="white--text submit-project"
                  color=""
                  data-test-id="btn-submit-project-info"
                  depressed=""
                  disabled="true"
                  name="button"
                  type="button"
                >
                  
                  
                
                </button-stub>
              </v-card-actions-stub>
            </v-card-stub>
          </v-col-stub>
        </v-row-stub>
      </v-container-stub>
    </v-form-stub>
  </v-card-stub>
</v-card-stub>
`;

exports[`AddProjectInfo.vue renders props when passed 1`] = `
<v-card-stub
  class="mx-auto outer-card"
  loaderheight="4"
  tag="div"
>
  <v-card-stub
    class="mx-auto"
    loaderheight="4"
    tag="div"
  >
    <v-app-bar-stub
      class="bc-subtitle"
      dark="true"
      extensionheight="48"
      src=""
      tag="header"
      tile="true"
      value="true"
    >
      <v-btn-stub
        activeclass=""
        aria-label="Back Button"
        icon="true"
        tag="button"
        type="button"
      >
        <v-icon-stub>
          mdi-arrow-left
        </v-icon-stub>
      </v-btn-stub>
       
      <v-toolbar-title-stub>
        
        
      
      </v-toolbar-title-stub>
       
      <v-spacer-stub />
    </v-app-bar-stub>
     
    <v-form-stub>
      <v-container-stub
        tag="div"
      >
        <v-row-stub
          dense="true"
          tag="div"
        >
          <v-col-stub
            cols="12"
            md="12"
            tag="div"
          >
            <v-card-stub
              class="pa-8 pt-6 ma-3"
              loaderheight="4"
              tag="div"
            >
              <v-card-subtitle-stub
                class="text-left bc-padding-left-0 page-info"
              />
               
              <input-stub
                data-test-id="input-org-name"
                helptext=""
                label=""
                outlined="true"
                rules="v => !!v || 'This field is required',v => (v || '').length >= len || 'Invalid character length, required ' + len,v => (v || '').length <= len || 'Invalid character length, max length ' + len"
                type="text"
                value=""
              />
               
              <input-stub
                data-test-id="input-project-name"
                helptext=""
                label=""
                outlined="true"
                rules="v => !!v || 'This field is required',v => (v || '').length >= len || 'Invalid character length, required ' + len,v => (v || '').length <= len || 'Invalid character length, max length ' + len"
                type="text"
                value=""
              />
               
              <textarea-stub
                counter=""
                data-test-id="text-project-description"
                helptext=""
                label=""
                outlined=""
                rules="v => !!v || 'This field is required'"
                type="text"
                value=""
              />
            </v-card-stub>
          </v-col-stub>
           
          <v-col-stub
            cols="12"
            tag="div"
          >
            <v-card-stub
              flat="true"
              loaderheight="4"
              tag="div"
            >
              <v-divider-stub />
               
              <v-card-actions-stub>
                <v-spacer-stub />
                 
                <!---->
                 
                <button-stub
                  class="white--text submit-project"
                  color=""
                  data-test-id="btn-submit-project-info"
                  depressed=""
                  disabled="true"
                  name="button"
                  type="button"
                >
                  
                  
                
                </button-stub>
              </v-card-actions-stub>
            </v-card-stub>
          </v-col-stub>
        </v-row-stub>
      </v-container-stub>
    </v-form-stub>
  </v-card-stub>
</v-card-stub>
`;
